hazelcast/src/main/java/com/hazelcast/instance/BuildInfoProvider.java

```diff
@@ -81,20 +84,42 @@ static void setJetProperties(Properties properties, BuildInfo buildInfo) {
     }
 
     private static Properties loadPropertiesFromResource(String resourceName) {
-        InputStream properties = BuildInfoProvider.class.getClassLoader().getResourceAsStream(resourceName);
+        ClassLoader classLoader = BuildInfoProvider.class.getClassLoader();
+        URL lastUrl = findLastResourceURL(resourceName, classLoader);
         Properties runtimeProperties = new Properties();
+        if (lastUrl == null) {
+            //apparently there is no resource matching this name
+            return runtimeProperties;
+        }
+        InputStream propertyStream = null;
         try {
-            if (properties != null) {
-                runtimeProperties.load(properties);
+            propertyStream = lastUrl.openStream();
+            if (propertyStream != null) {
+                runtimeProperties.load(propertyStream);
             }
         } catch (Exception ignored) {
             ignore(ignored);
         } finally {
-            closeResource(properties);
+            closeResource(propertyStream);
         }
         return runtimeProperties;
     }
 
+    private static URL findLastResourceURL(String resourceName, ClassLoader classLoader) {
```

Comment:
While this method uses this order, it's only a recommendation that implementations overriding this do the same.  Safest approach IMO is to use `findResource` via reflection.

```
@apiNote When overriding this method it is recommended that an
     * implementation ensures that any delegation is consistent with the {@link
     * #getResources(java.lang.String) getResources(String)} method.
```